src/test/java/teammates/test/cases/webapi/DeleteStudentActionTest.java

```diff
@@ -39,10 +39,10 @@ protected void testExecute() {
                 Const.ParamsNames.STUDENT_EMAIL, student1InCourse1.email,
         };
 
-        DeleteStudentAction a = getAction(submissionParams);
-        JsonResult r = getJsonResult(a);
+        DeleteStudentAction deleteStudentAction = getAction(submissionParams);
+        JsonResult getOutput = getJsonResult(deleteStudentAction);
 
-        assertEquals(HttpStatus.SC_OK, r.getStatusCode());
+        assertEquals(HttpStatus.SC_OK, getOutput.getStatusCode());
 
         ______TS("success: delete a student by id");
```

Comment:
Yup I added a few tests and found some potential NPEs (fixed).

How should the design be? We should throw a `404 not found` if a random email/id is given right? So that the user of the API is aware that nothing happened? Or should we fail silently? (the backend logic seems to be currently doing this)

Reply:
IMO, `DELETE` endpoints _should_ fail with a 404 if a non-existent resource is specified, but opinion is divided over this because the behavior in this case is not defined by the HTTP 1.1 spec. One argument for silent failure is that deletes should be idempotent because you just want to ensure that the resource with the specified details does not exist, so repeated calls or attempting to delete something non-existent should return a 200 since the entity with those details does not exist. Tl;dr it's use-case and philosophy-dependent.

For our case, we should fail deletes silently as per our storage component policy: https://github.com/TEAMMATES/teammates/blob/master/docs/design.md#policies-1

My bad for the confusion. The negative test cases should still stay but they should always assert for 200. This will let us catch deviants trying to break the storage policy. You'll need to undo the changes you made to the action.

Reply:
Okay no worries! Reverted accordingly